lib/game.rb

```diff
@@ -0,0 +1,240 @@
+require './lib/card'
+require './lib/deck'
+require './lib/player'
+require './lib/turn'
+
+
+
+class Game
+
+    attr_reader :player1,
+                :player2,
+                :deck1,
+                :deck2
+
+    def initialize 
+
+        
+
+        create_decks
+
+        
+        @player1 = Player.new("Mufasa", @deck1)
+        @player2 = Player.new("Skar", @deck2)
+        
+
+    end
+
+
+
+
+
+    def create_high_value(suits, suit)
+
+        suits << Card.new(suit, "Jack", 11)
+        suits << Card.new(suit, "Queen", 12)
+        suits << Card.new(suit, "King", 13)
+        suits << Card.new(suit, "Ace", 14)
+
+    end
+
+
+
+
+
+
+    def create_decks
+
+        rank_values = [2, 3, 4, 5, 6, 7, 8, 9, 10]
+
+        diamonds = []
+        hearts = []
+        spades = []
+        clubs = []
+
+        full_deck_cards = []
+
+
+        rank_values.each do |rank_value|
+
+            diamonds << Card.new(:diamond, "#{rank_value}", rank_value)
+
+            hearts << Card.new(:heart, "#{rank_value}", rank_value)
+
+            spades << Card.new(:spade, "#{rank_value}", rank_value)
+
+            clubs << Card.new(:club, "#{rank_value}", rank_value)
+            
+        end
+
+
+        create_high_value(diamonds, :diamond)
+        create_high_value(hearts, :heart)
+        create_high_value(spades, :spade)
+        create_high_value(clubs, :club)
+
+
+        full_deck_cards = diamonds + hearts + spades + clubs
+        full_deck_cards = full_deck_cards.shuffle
+        half_deck_1 = []
+        half_deck_2 = []
+
+
+        26.times do
+        half_deck_1 << full_deck_cards.shift
+        end
+
+        26.times do
+        half_deck_2 << full_deck_cards.shift
+        end
+
+        @deck1 = Deck.new(half_deck_1)
+        @deck2 = Deck.new(half_deck_2)
+
+    end
+
+
+
+
+
+    def welcome_message
+        print "\nWelcome to War (or Peace)! This game will be played with 52 cards.\nThe players today are #{@player1.name} and #{@player2.name}.\nType 'GO' to start the game!\n ------------------------------------------------------------------\n" 
+    end
+
+
+
+
+
+
+    def user_input
+        gets.chomp
+    end
+
+
+
+
+
+    def welcome_loop
+
+        welcome_message
+
+        loop do
+
+            if user_input != "GO"
+
+                welcome_message
+
+            else
+
+                break
+
+            end
+        
+        end
+
+    end
+
+
+
+
+
+
+    def start
```

Comment:
This method feels very long to me. Similar to how you pull your `welcome_message` method out to its own method in the `welcome_loop`, I wish you'd done more of that here so your `start` method would be a little more digestible. There's a lot going on and that leaves us prone to bugs.

Reply:
I actually realized that post submission when I heard about keeping methods to 7 lines at most (can't remember who I heard it from, I think it was Abdul?) later on, and how having more helper methods can allow us to test for each individual block of code and avoid running into the user input testing issue that I experienced.

Reply:
The rule of thumb is about 7 yeah. Essentially once our methods start getting longer than that, it's a sign that the method might be doing too much. It's not an absolute, just a trigger to take a second look. Sometimes it makes sense to have a couple longer methods for the sake of readability, for example.  But yes, helpers allows us to test things a little more granularly which can prevent bugs as well.